Sources/Sentry/SentryTracer.m

```diff
@@ -29,13 +33,13 @@ - (instancetype)initWithTransactionContext:(SentryTransactionContext *)transacti
                            waitForChildren:(BOOL)waitForChildren
 {
     if ([super init]) {
-        _rootSpan = [[SentrySpan alloc] initWithTracer:self context:transactionContext];
+        self.rootSpan = [[SentrySpan alloc] initWithTracer:self context:transactionContext];
         self.name = transactionContext.name;
-        _children = [[NSMutableArray alloc] init];
-        _hub = hub;
-        _isFinished = YES;
+        self.children = [[NSMutableArray alloc] init];
+        self.hub = hub;
+        self.isWaitingForChildren = NO;
```

Comment:
Basically this is the change? The rest is refactoring?
If CI is green, perhaps we need a test to make sure there's no regression in the future.

Reply:
Excellent that you point out to avoid regressions in the future. @brustolin covers that with https://github.com/getsentry/sentry-cocoa/pull/1144. @brustolin  and I agreed to merge https://github.com/getsentry/sentry-cocoa/pull/1150 without tests because this PR fixed a nasty bug and we wanted to have it in here ASAP.